pkgs/applications/misc/syncthingtray/default.nix

```diff
@@ -0,0 +1,56 @@
+{ mkDerivation
+, stdenv
+, lib
+, fetchFromGitHub
+, qtbase
+, qtwebengine
+, qtdeclarative
+, extra-cmake-modules
+, cpp-utilities
+, qtutilities
+, cmake
+, kio
+, plasma-framework
+, qttools
+, webviewSupport ? false
+, jsSupport ? false
+, enableKioPluginSupport ? false
+, enablePlasmoidSupport  ? false
+, systemdSupport ? true
+}:
+
+mkDerivation rec {
+  version = "0.9.1";
+  pname = "syncthingtray";
+
+  src = fetchFromGitHub {
+    owner = "Martchus";
+    repo = "syncthingtray";
+    rev = "v${version}";
+    sha256 = "0ijwpwlwwbfh9fdfbwz6dgi6hpmaav2jm56mzxm6as50iwnb59fx";
+  };
+
+  buildInputs = [ qtbase cpp-utilities qtutilities ]
+    ++ lib.optionals webviewSupport [ qtwebengine ]
+    ++ lib.optionals jsSupport [ qtdeclarative ]
+    ++ lib.optionals enableKioPluginSupport [ kio ]
+    ++ lib.optionals enablePlasmoidSupport [ extra-cmake-modules plasma-framework ]
+  ;
+
+  nativeBuildInputs = [ cmake qttools ];
+
+  cmakeFlags = lib.optionals (!enablePlasmoidSupport) ["-DNO_PLASMOID=ON"]
+    ++ lib.optionals (!enableKioPluginSupport) ["-DNO_FILE_ITEM_ACTION_PLUGIN=ON"]
+    ++ lib.optionals systemdSupport ["-DSYSTEMD_SUPPORT=ON"]
+    # See https://github.com/Martchus/syncthingtray/issues/42
+    ++ lib.optionals enablePlasmoidSupport ["-DQT_PLUGIN_DIR:STRING=${placeholder "out"}/lib/qt-5"]
```

Comment:
I have no experience with your build system so just ignore this if I'm wrong - but to me it looks like `-DQT_PLUGIN_DIR...` would only be added if one enables the Plasmoid support. However, the KIO plugin is also a plugin and most likely needs this as well. I'd just add `-DQT_PLUGIN_DIR...` unconditionally. Setting this variable will not cause any problems if no plugins are built after all.